inc/datahandlers/user.php

```diff
@@ -888,6 +888,20 @@ function verify_checkfields()
 	}
 
 	/**
+	 * Verifies if the selected timezone is valid.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_timezone()
+	{
+		$user = &$this->data;
+
+		$timezones = get_supported_timezones();
+
+		return array_key_exists($user['timezone'], $timezones);
```

Comment:
I am happy to change the function accordingly, but a lot of `verify` functions do not set an error. For example, `verify_referrer` and `verify_lastvisit` do not set an error, but they are setting a default value. I could move the handling of the default value to the `verify_timezone`, if that would make the code more consistent.